### PR TITLE
[Components with more than 1 service] LPS-188721 Refactor ModelListener components part II

### PR DESCRIPTION
**Pull request by MrJohn1911**
Ticket: https://liferay.atlassian.net/browse/LPS-188721

Previous pull: https://github.com/liferay-core-infra/liferay-portal/pull/4936

### CODE REVIEW

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:search

**Comment by catarineaguiar** [comment]
@MrJohn1911 regression found:

```
     [echo] diff --git a/modules/dxp/apps/search-experiences/search-experiences-api/src/main/java/com/liferay/search/experiences/service/SXPElementLocalService.java b/modules/dxp/apps/search-experiences/search-experiences-api/src/main/java/com/liferay/search/experiences/service/SXPElementLocalService.java
     [echo] index 495e4d537ec7..7aa6ef091ae7 100644
     [echo] --- a/modules/dxp/apps/search-experiences/search-experiences-api/src/main/java/com/liferay/search/experiences/service/SXPElementLocalService.java
     [echo] +++ b/modules/dxp/apps/search-experiences/search-experiences-api/src/main/java/com/liferay/search/experiences/service/SXPElementLocalService.java
     [echo] @@ -106,8 +106,6 @@ public interface SXPElementLocalService
     [echo]  	@Transactional(enabled = false)
     [echo]  	public SXPElement createSXPElement(long sxpElementId);
     [echo]  
     [echo] -	public void importSXPElements(Company company) throws PortalException;
     [echo] -
     [echo]  	public void deleteCompanySXPElements(long companyId) throws PortalException;
     [echo]  
     [echo]  	/**
     [echo] @@ -316,6 +314,8 @@ public interface SXPElementLocalService
     [echo]  	@Transactional(propagation = Propagation.SUPPORTS, readOnly = true)
     [echo]  	public int getSXPElementsCount();
     [echo]  
     [echo] +	public void importSXPElements(Company company) throws PortalException;
     [echo] +
     [echo]  	@Indexable(type = IndexableType.REINDEX)
     [echo]  	public SXPElement updateStatus(long userId, long sxpElementId, int status)
     [echo]  		throws PortalException;
     [echo] diff --git a/modules/dxp/apps/search-experiences/search-experiences-api/src/main/java/com/liferay/search/experiences/service/SXPElementLocalServiceUtil.java b/modules/dxp/apps/search-experiences/search-experiences-api/src/main/java/com/liferay/search/experiences/service/SXPElementLocalServiceUtil.java
     [echo] index 65caf5563038..42d76b5f3043 100644
     [echo] --- a/modules/dxp/apps/search-experiences/search-experiences-api/src/main/java/com/liferay/search/experiences/service/SXPElementLocalServiceUtil.java
     [echo] +++ b/modules/dxp/apps/search-experiences/search-experiences-api/src/main/java/com/liferay/search/experiences/service/SXPElementLocalServiceUtil.java
     [echo] @@ -358,6 +358,13 @@ public class SXPElementLocalServiceUtil {
     [echo]  		return getService().getSXPElementsCount();
     [echo]  	}
     [echo]  
     [echo] +	public static void importSXPElements(
     [echo] +			com.liferay.portal.kernel.model.Company company)
     [echo] +		throws PortalException {
     [echo] +
     [echo] +		getService().importSXPElements(company);
     [echo] +	}
     [echo] +
     [echo]  	public static SXPElement updateStatus(
     [echo]  			long userId, long sxpElementId, int status)
     [echo]  		throws PortalException {
     [echo] diff --git a/modules/dxp/apps/search-experiences/search-experiences-api/src/main/java/com/liferay/search/experiences/service/SXPElementLocalServiceWrapper.java b/modules/dxp/apps/search-experiences/search-experiences-api/src/main/java/com/liferay/search/experiences/service/SXPElementLocalServiceWrapper.java
     [echo] index af22de0b185c..8e9e430f0120 100644
     [echo] --- a/modules/dxp/apps/search-experiences/search-experiences-api/src/main/java/com/liferay/search/experiences/service/SXPElementLocalServiceWrapper.java
     [echo] +++ b/modules/dxp/apps/search-experiences/search-experiences-api/src/main/java/com/liferay/search/experiences/service/SXPElementLocalServiceWrapper.java
     [echo] @@ -14,8 +14,6 @@
     [echo]  
     [echo]  package com.liferay.search.experiences.service;
     [echo]  
     [echo] -import com.liferay.portal.kernel.exception.PortalException;
     [echo] -import com.liferay.portal.kernel.model.Company;
     [echo]  import com.liferay.portal.kernel.service.ServiceWrapper;
     [echo]  import com.liferay.portal.kernel.service.persistence.BasePersistence;
     [echo]  
     [echo] @@ -96,11 +94,6 @@ public class SXPElementLocalServiceWrapper
     [echo]  		return _sxpElementLocalService.createSXPElement(sxpElementId);
     [echo]  	}
     [echo]  
     [echo] -	@Override
     [echo] -	public void importSXPElements(Company company) throws PortalException {
     [echo] -		_sxpElementLocalService.importSXPElements(company);
     [echo] -	}
     [echo] -
     [echo]  	@Override
     [echo]  	public void deleteCompanySXPElements(long companyId)
     [echo]  		throws com.liferay.portal.kernel.exception.PortalException {
     [echo] @@ -412,6 +405,14 @@ public class SXPElementLocalServiceWrapper
     [echo]  		return _sxpElementLocalService.getSXPElementsCount();
     [echo]  	}
     [echo]  
     [echo] +	@Override
     [echo] +	public void importSXPElements(
     [echo] +			com.liferay.portal.kernel.model.Company company)
     [echo] +		throws com.liferay.portal.kernel.exception.PortalException {
     [echo] +
     [echo] +		_sxpElementLocalService.importSXPElements(company);
     [echo] +	}
     [echo] +
     [echo]  	@Override
     [echo]  	public com.liferay.search.experiences.model.SXPElement updateStatus(
     [echo]  			long userId, long sxpElementId, int status)
[stopwatch] [run.batch.test.action: 28:25.093 sec]
     [echo] .
     [echo] 
     [echo] Detected build services changes. See above build services log for more
     [echo] information. Make sure to commit in all build services results.
```
full log [here](https://testray.liferay.com/reports/production/logs/2023-07/test-1-32/test-portal-acceptance-pullrequest(master)/821/service-builder-jdk8/0/0/jenkins-console.txt.gz?authuser=0)

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:search

**Comment by dantewang** [comment]
@MrJohn1911 Please exclude the manual changes on the generated files in https://github.com/liferay-core-infra/liferay-portal/pull/5014/commits/7068fa91340a3b89f24955ab352ef1792203829c as long as they are re-generated by Service Builder.

File: modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/service/impl/SXPElementLocalServiceImpl.java [hunk 1]
```diff
@@ -133,6 +152,49 @@ public List<SXPElement> getSXPElements(long companyId, boolean readOnly) {
...
+		for (SXPElement sxpElement :
+				getSXPElements(company.getCompanyId(), true)) {
+
+			externalReferenceCodes.add(sxpElement.getExternalReferenceCode());
+		}
+
+		for (com.liferay.search.experiences.rest.dto.v1_0.SXPElement
+				sxpElement : _getSXPElements()) {
+
+			if ((!FeatureFlagManagerUtil.isEnabled("LPS-122920") &&
```
Comment (dantewang) [inline modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/service/impl/SXPElementLocalServiceImpl.java:168]: @MrJohn1911 Please check if we can move this feature flag check to the beginning of this method.

**Review by dantewang — CHANGES_REQUESTED** [review]

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by Kvale1733** [comment]
@MrJohn1911 Regression found.

1. com.liferay.search.experiences.internal.model.listener.test.SXPElementLocalServiceImplTest
```
 [exec] 
     [exec] com.liferay.search.experiences.internal.model.listener.test.SXPElementLocalServiceImplTest > testSXPElements STARTED
     [exec] 
     [exec] com.liferay.search.experiences.internal.model.listener.test.SXPElementLocalServiceImplTest > testSXPElements FAILED
     [exec]     java.lang.NullPointerException
     [exec]     	at com.liferay.search.experiences.internal.model.listener.test.SXPElementLocalServiceImplTest.testSXPElements(SXPElementLocalServiceImplTest.java:50)
     [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
     [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
     [exec]     	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
     [exec]     	at java.lang.reflect.Method.invoke(Method.java:498)
     [exec]     	at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable$3.evaluate(TestExecutorRunnable.java:353)
     [exec]     	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:99)
     [exec]     	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:99)
```
[Log](https://testray.liferay.com/reports/production/logs/2023-07/test-1-26/test-portal-acceptance-pullrequest(master)/10621/modules-integration-mysql57-jdk8/0/11/jenkins-console.txt.gz?authuser=0)

2) com.liferay.search.experiences.internal.blueprint.test.SXPBlueprintSearchResultTest
```
[exec] com.liferay.search.experiences.internal.blueprint.test.SXPBlueprintSearchResultTest > testBoostAssetType FAILED
     [exec]     java.util.NoSuchElementException
     [exec]     	at com.liferay.search.experiences.internal.blueprint.test.SXPBlueprintSearchResultTestUtil.getElementInstancesJSON(SXPBlueprintSearchResultTestUtil.java:76)
     [exec]     	at com.liferay.search.experiences.internal.blueprint.test.SXPBlueprintSearchResultTest._updateElementInstancesJSON(SXPBlueprintSearchResultTest.java:2459)
     [exec]     	at com.liferay.search.experiences.internal.blueprint.test.SXPBlueprintSearchResultTest.testBoostAssetType(SXPBlueprintSearchResultTest.java:197)
     [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
     [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
     [exec]     	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
     [exec]     	at java.lang.reflect.Method.invoke(Method.java:498)
     [exec]     	at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable$3.evaluate(TestExecutorRunnabl
```
[Log](https://testray.liferay.com/reports/production/logs/2023-07/test-1-26/test-portal-acceptance-pullrequest(master)/10621/modules-integration-mysql57-jdk8/0/11/jenkins-console.txt.gz?authuser=0)

3) Blueprint elements no present. Probably why number 2 fails
Steps to reproduce: 
1. Go to Applicaions > Search Experiences > Blueprints
2. Click Elements Tab

Expected Results. Several elements should be present 
Actual Results: There are no elements at all

Should be something like this 
sc

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:search

**Comment by catarineaguiar** [comment]
@MrJohn1911 regression found:

Query Elements are not shown on Bluprints edit.

Steps to reproduce:

1. Go to Applications > search experiences > Blueprints
2. Add a new blueprint
3. Click to add a query element

**Actual result:**
Elements in the "Add Query Elements" panel are not shown. Only custom.
image

Expected result:
All elements in "Add Query Elements" panel are shown.
image

**Pull request closed without merging**